x-pack/plugins/serverless_search/public/application/components/overview_panels/select_client.tsx

```diff
@@ -0,0 +1,174 @@
+}
+export const SelectClientPanel: React.FC<SelectClientProps> = ({
+  setSelectedLanguage,
+  languages,
+  selectedLanguage,
+}) => {
+  const { http } = useKibanaServices();
+  const { euiTheme } = useEuiTheme();
+
+  const panelItems = languages.map((language) => (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem grow={false}>
+        <EuiPanel
+          hasBorder
+          borderRadius="m"
+          onClick={() => setSelectedLanguage(language)}
+          grow={false}
+          paddingSize="m"
+          css={
+            selectedLanguage === language
+              ? css`
+                  border: 1px solid ${euiTheme.colors.primary};
+                `
+              : css`
+                  border: 1px solid ${euiTheme.colors.lightShade};
+                `
+          }
+          color={selectedLanguage === language ? 'primary' : 'plain'}
+        >
+          <EuiFlexGroup direction="column" gutterSize="l" justifyContent="center">
+            <EuiFlexItem>
+              <EuiImage
+                alt=""
+                src={http.basePath.prepend(`/plugins/${PLUGIN_ID}/assets/${language.iconType}`)}
+                height="32px"
+                width="32px"
+              />
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiText
+                textAlign="center"
+                color={selectedLanguage === language ? 'default' : 'subdued'}
+              >
+                <h5>{language.name}</h5>
+              </EuiText>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiPanel>
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  ));
+
+  return (
+    <OverviewPanel
+      description={
+        <FormattedMessage
+          id="xpack.serverlessSearch.selectClient.description"
+          defaultMessage="Elastic builds and maintains clients in several popular languages and our community has contributed many more. Select your favorite language client or dive into the {console} to get started."
+          values={{
+            console: (
+              <EuiLink href={http.basePath.prepend(`/app/dev_tools#/console`)}>
+                {i18n.translate('xpack.serverlessSearch.selectClient.description.console.link', {
+                  defaultMessage: 'Console',
+                })}
+              </EuiLink>
+            ),
+          }}
+        />
+      }
+      leftPanelContent={
+        <>
+          <EuiFlexGroup gutterSize="l" direction="column">
+            <EuiFlexItem>
+              <EuiText size="s">
+                <strong>
+                  {i18n.translate('xpack.serverlessSearch.selectClient.heading', {
+                    defaultMessage: 'Choose one',
+                  })}
+                </strong>
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiFlexGrid
+                gutterSize="l"
+                direction="row"
+                columns={panelItems.length > 3 ? 4 : 3}
```

Comment:
Just realized that Grid may not be the best solution here. Grid can have maximum of 4 columns. As per the designs, we might support more than 4 clients and need everything in a single row. Hence working on replacing with Flex group 
@TattdCodeMonkey